src/main/java/org/apache/sysml/runtime/instructions/gpu/context/GPUObject.java

```diff
@@ -301,6 +301,10 @@ public CSRPointer getSparseMatrixCudaPointer() {
 	 * @throws DMLRuntimeException ?
 	 */
 	public void setSparseMatrixCudaPointer(CSRPointer sparseMatrixPtr) throws DMLRuntimeException {
+		if (this.jcudaSparseMatrixPtr != null) {
```

Comment:
where did you encounter this issue ? if we are doing shallow copy, then we should hold back references. For example: [GPUPointer which I was planning to add for left indexing](https://github.com/niketanpansare/systemml/blob/bf796a9c2d346807441b518bce3eb6375c426001/src/main/java/org/apache/sysml/runtime/instructions/gpu/context/GPUPointer.java#L39).

Reply:
I've put this here as a defensive check and the log the message so that I can go check for what went wrong.

Reply:
ok, then in this case, I would recommend removing `deallocate` as shallow copy is a valid optimization and `deallocate` will make it difficult to debug. Instead, it makes sense to either put an exception or just warning.